### PR TITLE
Add `warnings` to magnitude and angle_between

### PR DESCRIPTION
**Pull request by sidhu1012**
#### References to other Issues or PRs

#21729 

#### Brief description of what is fixed or changed
Added a `warnings` section to `.magnitude()` and `.angle_between()` to inform user.

#### Release Notes

* physics.vector
    * `Vector.angle_between()` has a new section `Warnings`.
    * `Vector.magnitude()` has a new section `Warnings`.

### CODE REVIEW

File: sympy/physics/vector/vector.py [hunk 1]
```diff
@@ -678,7 +678,17 @@ def subs(self, *args, **kwargs):
         return Vector(d)
 
     def magnitude(self):
-        """Returns the magnitude (Euclidean norm) of self."""
+        """Returns the magnitude (Euclidean norm) of self.
+
+        Warnings
+        ========
```
Comment (moorepants) [inline sympy/physics/vector/vector.py:684]: I think the heading "Warnings" is supposed to explain actual warnings that are returned. Just make this "Notes".

Comment (sidhu1012) [reply sympy/physics/vector/vector.py:684]: The `Warns` section is for actual warnings, `Warnings` is the one for this.

Comment (moorepants) [reply sympy/physics/vector/vector.py:684]: I'm wrong about that:

https://numpydoc.readthedocs.io/en/latest/format.html#warns

**Comment by moorepants** [comment]
LGTM

**Pull request merged**